application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -74,6 +74,9 @@
   ## editor doesn't use standard text fields for entering the labels for the various shapes the diagram is made of)
   #set ($keyboardShortcutsEnabled = false)
   #set ($discard = $xwiki.ssx.use('Diagram.DiagramSheet'))
+  ## Make sure that the version loaded with RequireJS is not a cached one. See XWIKI-12981: Add the document version
+  ## of skin extensions in the generated URLs.
+  #set ($discard = $xwiki.jsx.use('Diagram.DiagramSheet'))
```

Comment:
I'm not sure about this. You're now loading ``Diagram.DiagramSheet`` twice: once here and once with RequireJS (I doubt the URLs match, so the browser sees them as two different resources). We already pass some query string parameters when loading with RequireJS, couldn't we add the document or XWiki version also, to make sure the cache is invalidated when upgrading XWiki?

Reply:
I added the diagram version as a query param, since the page should be modified only at upgrade